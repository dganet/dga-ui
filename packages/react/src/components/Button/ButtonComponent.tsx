import React from 'react'

import { ButtonStyled } from '@dga-ui/styles'
import { styled } from '@dga-ui/web'
import { ButtonProps } from './ButtonComponent.types'

const StyledButton = styled('button', ButtonStyled)

export const ButtonComponent = ({
  children,
  buttonSize = 'large',
  buttonType = 'primary',
  onlyIcon = false,
  disabled = false,
  icon: Icon
}: ButtonProps) => {
  return (
    <StyledButton
      onlyIcon={onlyIcon}
      disabled={disabled}
      size={buttonSize}
      colors={buttonType}
    >
      {Icon && <Icon />} <label>{children}</label>
    </StyledButton>
  )
}

ButtonComponent.displayName = 'Button'
