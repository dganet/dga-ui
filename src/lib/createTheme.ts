import { defaultTheme } from '../styles/defaultTheme'

export const createTheme = (theme: typeof defaultTheme) => {
    return {
        ...defaultTheme,
        colors: {
            ...defaultTheme.colors,
            ...theme.colors
        },

        name: theme.name || defaultTheme.name
    } as typeof defaultTheme
}
