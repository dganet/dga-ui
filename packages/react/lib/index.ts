import { ButtonComponent } from "./components/Button/ButtonComponent";

export {
  ButtonComponent as Button,
}