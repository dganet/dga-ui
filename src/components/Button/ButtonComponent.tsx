import { useTheme } from '@hooks/useTheme'
import React from 'react'

import { Container } from './ButtonComponent.styles'
import { ButtonProps } from './ButtonComponent.types'

const ButtonComponent: React.FC<ButtonProps> = ({
    children,
    buttonSize = 'large',
    buttonType = 'primary',
    onlyIcon = false,
    icon: Icon
}) => {
    const theme = useTheme()
    return (
        <Container
            theme={theme}
            buttonSize={buttonSize}
            buttonType={buttonType}
            onlyIcon={!!!children}
        >
            {Icon && <Icon />} <label>{children}</label>
        </Container>
    )
}

export default ButtonComponent
