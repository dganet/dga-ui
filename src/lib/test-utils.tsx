import { render, RenderOptions } from '@testing-library/react'
import React, { ReactElement, FC } from 'react'
import { defaultTheme } from '../styles/defaultTheme'
import { CustomThemeProvider } from '../hooks/useTheme/ThemeContext'

const Providers: FC = ({ children }) => {
    return (
        <CustomThemeProvider theme={defaultTheme}>
            {children}
        </CustomThemeProvider>
    )
}

/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
