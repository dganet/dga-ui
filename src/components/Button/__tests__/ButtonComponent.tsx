import { render, screen } from '../../../lib/test-utils'

import ButtonComponent from '../ButtonComponent'

describe('<ButtonComponent />', () => {
    it('should render the component', () => {
        const { container } = render(<ButtonComponent />)

        expect(container.firstChild).toMatchSnapshot()
    })
})
