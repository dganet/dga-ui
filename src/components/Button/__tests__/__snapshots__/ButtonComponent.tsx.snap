// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComponent /> should render the component 1`] = `
<button
  class="sc-bdfBQB fWrzJR"
>
   
  <label />
</button>
`;
